Lib/argparse.py

```diff
@@ -2553,11 +2557,30 @@ def _get_value(self, action, arg_string):
         return result
 
     def _check_value(self, action, value):
+        if not action.choices and isinstance(action.choices, list):
+            msg = 'Either add options in choices array or remove it'
+            raise ArgumentError(action, msg)
+
         # converted value must be one of the choices (if specified)
         if action.choices is not None and value not in action.choices:
-            args = {'value': value,
-                    'choices': ', '.join(map(repr, action.choices))}
+            args = {
+                'value': value,
+                'choices': ', '.join(map(repr, action.choices)),
+            }
             msg = _('invalid choice: %(value)r (choose from %(choices)s)')
+
+            if self.suggest_on_error:
+                try:
+                    import difflib
+                    closest_choice = difflib.get_close_matches(value, action.choices, 1)
```

Comment:
Does it work with non-strings? Add test with `type=int`.

Reply:
I'm struggling to come up with a rationale for why this should support types other than string. I think that this is most useful when this is of type string. I will add a guard for this and document this limitation.

Reply:
We should test that this code works in some way or another if `choices` is not a collection of strings. Failing due to internal error would be bad.